### DISCUSSION TITLE
Count rows for view

### DISCUSSION THREAD

**Discussion by devopg** [MaterializeInc/materialize#12791] [Q&A] [answered]
Hello. Dont understand how can i select count() for view ? 
I try 
`SELECT count(*) FROM public.market_orders;`
buy get error:
![image](https://user-images.githubusercontent.com/61819835/171276554-ec8569be-e48b-44d6-9058-4fe0576f2821.png)

Okey, i try use `AS OF `

`SELECT count(*) FROM public.market_orders as of ts;`
buy got error about column dont exists
![image](https://user-images.githubusercontent.com/61819835/171276697-edd26f26-d0e3-4739-b560-754c651191c6.png)

okey i make query for take table inforamtion 
```
SELECT
    column_name,
    data_type
FROM
    information_schema.columns
WHERE
    table_name = 'market_orders';
```
![image](https://user-images.githubusercontent.com/61819835/171277154-0938af50-ed4a-4724-abcc-0359331f756a.png)

column ts exists and have timestamp type

**My other try:**
`SELECT * FROM public.market_orders as of 1`
![image](https://user-images.githubusercontent.com/61819835/171279843-dac09120-dc80-402a-a90b-93e16b138b2d.png)

**Comment by bobbyiliev** [comment, marked as answer]
Hi there,

It looks like that the `market_orders` view is not materialized. You can verify this with the following query:

```sql
SHOW FULL VIEWS;
```

You can almost think the NON-materialized views as a reusable template to be used in other materialized views. The NON-materialized views do not actually process any data.

What you can do is create a materialized view with the following:

```sql
CREATE MATERIALIZED VIEW market_orders_count AS SELECT count(*) FROM public.market_orders;
```

The above will create a materialized view called `market_orders_count` which will contain the result of your query and the results will be incrementally updated as the data changes.

For more information on materialized views and how they work, I would suggest the following documentation:

https://materialize.com/docs/sql/create-materialized-view/

In case that you want to learn more about the non-materialized views, you can check out the following:

https://materialize.com/docs/sql/create-view/

Hope that this helps and let me know if you have any questions!

**Comment by devopg** [comment]
yes its VIEW (non mater).
But how i can ceate temp query for show result for next make  MATERIALIZED VIEW ? 
Or i should 
1) create  MATERIALIZED VIEW
2) select result 
3) if wrong remove query
4) goto 1.

**Reply by bobbyiliev** [reply]
There are a couple of ways to do it:

- Use `TAIL`:

```sql
COPY ( TAIL ( SELECT count(*) FROM public.market_orders; ) ) TO STDOUT;
```

Basically, you can use tail to pass any query which will start a dataflow.

- Create a temp view so that it is not persisted and will be destroyed at the end of your session:

```sql
CREATE TEMPORARY MATERIALIZED VIEW market_orders_count_temp AS SELECT count(*) FROM public.market_orders;
```

Hope that this helps!

**Comment by devopg** [comment]
@bobbyiliev 
One more question. I watched video "Materialize Hack Day" on youtube from Materialize channel and saw how "Anreas Fuch" create source from kafka, create VIEW (non mat) and use "select count() from view" and its work. Is it removed functionality? 

![image](https://user-images.githubusercontent.com/61819835/171820500-a4024914-70ed-440a-9f2a-60cdd32b8225.png)

with time:
https://youtu.be/Kv9FCfZjgy0?t=3623

**Reply by bobbyiliev** [reply]
Hi there @devopg 

This is because in the demo the source is actually materialized. So the data is already ingested.

You can take a look a the DDL statements here:

https://github.com/antifuchs/mz-certificate-transparency/blob/master/SCHEMA.md

Hope that this helps!
